pallets/manta-pay/src/lib.rs

```diff
@@ -806,13 +808,18 @@ where
     {
         sources
             .map(move |(account_id, withdraw)| {
-                FungibleLedger::<T>::can_withdraw(asset_id.0, &account_id, withdraw.0)
-                    .map(|_| WrapPair(account_id.clone(), withdraw))
-                    .map_err(|_| InvalidSourceAccount {
-                        account_id,
-                        asset_id,
-                        withdraw,
-                    })
+                FungibleLedger::<T>::can_reduce_by_amount(
+                    asset_id.0,
+                    &account_id,
+                    withdraw.0,
+                    ExistenceRequirement::KeepAlive,
+                )
+                .map(|_| WrapPair(account_id.clone(), withdraw))
+                .map_err(|_| InvalidSourceAccount {
+                    account_id,
+                    asset_id,
+                    withdraw,
+                })
```

Comment:
It doesn't relate to this PR, I kind don't like returning `Vec` directly. Perhaps iterator would be better. Because iterator behaves laziness, but `collect` will allocate memory on head directly.

Reply:
@bhgomes